Orange/widgets/data/owgroupby.py

```diff
@@ -0,0 +1,490 @@
+                    # automatically but we need a callback call so we emit it
+                    self.stateChanged.emit(Qt.PartiallyChecked)
+            else:  # self.checkState() == Qt.Unchecked
+                # if unchecked: check if all can be checked else partially check
+                self.setCheckState(
+                    Qt.Checked if can_be_applied_all else Qt.PartiallyChecked
+                )
+
+
+@contextmanager
+def block_signals(widget):
+    widget.blockSignals(True)
+    yield
+    widget.blockSignals(False)
+
+
+class OWGroupBy(OWWidget, ConcurrentWidgetMixin):
+    name = "Group by"
+    description = ""
+    icon = "icons/GroupBy.svg"
+    keywords = ["aggregate", "group by"]
+
+    class Inputs:
+        data = Input("Data", Table, doc="Input data table")
+
+    class Outputs:
+        data = Output("Data", Table, doc="Aggregated data")
+
+    class Error(OWWidget.Error):
+        unexpected_error = Msg("{}")
+
+    settingsHandler = DomainContextHandler()
+
+    gb_attrs: List[Variable] = ContextSetting([])
+    aggregations: Dict[Variable, Set[str]] = ContextSetting({})
+    auto_commit: bool = Setting(True)
+
+    def __init__(self):
+        super().__init__()
+        ConcurrentWidgetMixin.__init__(self)
+
+        self.data = None
+        self.result = None
+
+        self.gb_attrs_model = DomainModel(
+            valid_types=(DiscreteVariable, ContinuousVariable, StringVariable),
+            separators=False,
+        )
+        self.agg_table_model = VarTableModel(self)
+        self.agg_checkboxes = {}
+
+        self.__init_control_area()
+        self.__init_main_area()
+
+    def __init_control_area(self) -> None:
+        """Init all controls in the control area"""
+        box = gui.vBox(self.controlArea, "Group by")
+        self.gb_attrs_view = AggregateListViewSearch(
+            selectionMode=QListView.ExtendedSelection
+        )
+        self.gb_attrs_view.setModel(self.gb_attrs_model)
+        self.gb_attrs_view.selectionModel().selectionChanged.connect(self.__gb_changed)
+        box.layout().addWidget(self.gb_attrs_view)
+
+        gui.auto_send(self.buttonsArea, self, "auto_commit")
+
+    def __init_main_area(self) -> None:
+        """Init all controls in the main area"""
+        # aggregation table
+        self.agg_table_view = tableview = QTableView()
+        tableview.setModel(self.agg_table_model)
+        tableview.setSelectionBehavior(QAbstractItemView.SelectRows)
+        tableview.selectionModel().selectionChanged.connect(self.__rows_selected)
+        tableview.horizontalHeader().setSectionResizeMode(QHeaderView.Stretch)
+
+        vbox = gui.vBox(self.mainArea, " ")
+        vbox.layout().addWidget(tableview)
+
+        # aggregations checkboxes
+        grid_layout = QGridLayout()
+        gui.widgetBox(self.mainArea, orientation=grid_layout, box="Aggregations")
+        itm_col = ceil(len(AGGREGATIONS) / 3)
```

Comment:
Interestingly, I'd write `(len(AGGREGATIONS) + 2) // 3` here. Always.

This is just an observation, not a suggestion. I guess it shows I'm old-school. My first CPUs didn't have floats. :) (Though: the difference is that `ceil` returns a `float`, while I get an `int`.)